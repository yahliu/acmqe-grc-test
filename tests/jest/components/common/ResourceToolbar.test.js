/*******************************************************************************
 * Licensed Materials - Property of IBM
 * (c) Copyright IBM Corporation 2018, 2019. All Rights Reserved.
 *
 * Note to U.S. Government Users Restricted Rights:
 * Use, duplication or disclosure restricted by GSA ADP Schedule
 * Contract with IBM Corp.
 *******************************************************************************/
/* Copyright (c) 2020 Red Hat, Inc. */
'use strict'

import React from 'react'
import { ResourceToolbar } from '../../../../src-web/components/common/ResourceToolbar'
import renderer from 'react-test-renderer'
import GrcApolloClient from '../../../../lib/client/apollo-client'
import { ApolloProvider } from 'react-apollo'

const availableFilters = {
  'standards': {
    'name': 'Standards',
    'availableSet': new Set([
      'NIST',
      'FISMA',
      'PCI',
      'HIPAA'
    ])
  },
  'categories': {
    'name': 'Categories',
    'availableSet': new Set([
      'System And Information Integrity',
      'System And Communications Protections'
    ])
  },
  'controls': {
    'name': 'Controls',
    'availableSet': new Set([
      'VA',
      'Secret Encryption',
      'Cert Manager',
      'Mutation Advisor'
    ])
  },
  'type': {
    'name': 'Type',
    'availableSet': new Set([
      'Enforce',
      'Inform only'
    ])
  },
  'severity': {
    'availableSet': new Set([])
  }
}
const location = {
  'pathname': '/multicloud/policies/policy/cluster1/1569249226915-policy-test',
  'search': '',
  'hash': '',
  'key': 'q1uagn'
}

describe('ResourceToolbar component', () => {
  it('renders with active filters as expected', () => {
    const activeFilters = {
      'standards': new Set([
        'HIPAA'
      ])
    }
    const refreshControl = {
      'reloading': false,
      'refreshCookie': 'grc-refresh-interval-cookie',
      'timestamp': 'Tue Sep 24 2019 09:56:26 GMT-0400 (Eastern Daylight Time)'
    }
    const updateFilterSpy = jest.fn()
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <ResourceToolbar
          updateActiveFilters={updateFilterSpy}
          filterViewOpen={true}
          location={location}
          refreshControl={refreshControl}
          activeFilters={activeFilters}
          availableFilters={availableFilters}
        />
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
    const toolbarComponent = component.root.findByType(ResourceToolbar)
    // Test toggling the filter panel
    const setStateSpy = jest.fn()
    toolbarComponent.instance.setState = setStateSpy
    toolbarComponent.instance.toggleFilterModelPress({key: 'Enter'})
    expect(setStateSpy.mock.calls).toHaveLength(1)
    // Test toggling an active filter
    toolbarComponent.instance.updateActiveFilter('standards', 'HIPAA', false)
    expect(updateFilterSpy).toHaveBeenCalledWith({ standards: new Set() })
  })
  it('renders with no filters as expected', () => {
    const component = renderer.create(
      <ApolloProvider client={GrcApolloClient.getGrcClient()}>
        <ResourceToolbar
          updateActiveFilters={jest.fn()}
          filterViewOpen={false}
          location={location}
          refreshControl={{}}
          activeFilters={{}}
          availableFilters={{}}
        />
      </ApolloProvider>
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
})
