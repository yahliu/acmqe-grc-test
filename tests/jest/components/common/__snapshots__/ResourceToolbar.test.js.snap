// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourceToolbar component renders with active filters as expected 1`] = `
<div
  className="resource-toolbar"
  id="resource-toolbar"
>
  <div
    className="resource-toolbar-container"
  >
    <div
      className="resource-toolbar-buttons"
    >
      <div
        className="refresh-time-selection"
      >
        <div
          aria-label="Refresh"
          className="button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
          title="Refresh"
        >
          <svg
            className="button-icon"
          >
            <use
              href="#diagramIcons_autoRefresh"
            />
          </svg>
        </div>
        <div
          aria-label="Choose refresh"
          className="bx--dropdown selection bx--list-box"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex={null}
        >
          <div
            aria-controls={null}
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="Choose refresh"
            aria-owns={null}
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <span
              className="bx--list-box__label"
              htmlFor="downshift-0-input"
            >
              Refresh every {0}s
            </span>
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                fillRule="evenodd"
                focusable="false"
                height="5"
                role="img"
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="resource-filter-view"
    style={
      Object {
        "height": 771,
      }
    }
  >
    <h3
      className="filterHeader"
    >
      <svg
        className="titleIcon"
      >
        <use
          href="#diagramIcons_filter"
        />
      </svg>
      <div
        className="titleText"
      >
        Filter by
      </div>
      <svg
        alt="Close filter view"
        aria-label="Close filter view"
        className="closeIcon"
        fillRule="evenodd"
        height="10"
        name="icon--close"
        onClick={[Function]}
        onKeyPress={[Function]}
        role="button"
        tabIndex={0}
        viewBox="0 0 10 10"
        width="10"
      >
        <title>
          Close filter view
        </title>
        <path
          d="M6.32 5L10 8.68 8.68 10 5 6.32 1.32 10 0 8.68 3.68 5 0 1.32 1.32 0 5 3.68 8.68 0 10 1.32 6.32 5z"
        />
      </svg>
    </h3>
    <div
      className="filter-sections-container"
      style={
        Object {
          "height": 688,
          "overflow": "hidden",
          "position": "relative",
          "width": 260,
        }
      }
    >
      <div
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "bottom": 0,
            "left": 0,
            "marginBottom": -17,
            "marginRight": 0,
            "overflow": "scroll",
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <div
          className="filter-section"
        >
          <div
            className="filter-section-title"
          >
            Standards
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="standardsall"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="standardsall"
              title={null}
            >
              <span
                className=""
              >
                All
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="standardsFISMA"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="standardsFISMA"
              title={null}
            >
              <span
                className=""
              >
                FISMA
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={true}
              className="bx--checkbox"
              id="standardsHIPAA"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="standardsHIPAA"
              title={null}
            >
              <span
                className=""
              >
                HIPAA
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="standardsNIST"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="standardsNIST"
              title={null}
            >
              <span
                className=""
              >
                NIST
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="standardsPCI"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="standardsPCI"
              title={null}
            >
              <span
                className=""
              >
                PCI
              </span>
            </label>
          </div>
        </div>
        <div
          className="filter-section"
        >
          <div
            className="filter-section-title"
          >
            Categories
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={true}
              className="bx--checkbox"
              id="categoriesall"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="categoriesall"
              title={null}
            >
              <span
                className=""
              >
                All
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="categoriesSystem And Communications Protections"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="categoriesSystem And Communications Protections"
              title={null}
            >
              <span
                className=""
              >
                System And Communications Protections
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="categoriesSystem And Information Integrity"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="categoriesSystem And Information Integrity"
              title={null}
            >
              <span
                className=""
              >
                System And Information Integrity
              </span>
            </label>
          </div>
        </div>
        <div
          className="filter-section"
        >
          <div
            className="filter-section-title"
          >
            Controls
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={true}
              className="bx--checkbox"
              id="controlsall"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="controlsall"
              title={null}
            >
              <span
                className=""
              >
                All
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="controlsCert Manager"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="controlsCert Manager"
              title={null}
            >
              <span
                className=""
              >
                Cert Manager
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="controlsMutation Advisor"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="controlsMutation Advisor"
              title={null}
            >
              <span
                className=""
              >
                Mutation Advisor
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="controlsSecret Encryption"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="controlsSecret Encryption"
              title={null}
            >
              <span
                className=""
              >
                Secret Encryption
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="controlsVA"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="controlsVA"
              title={null}
            >
              <span
                className=""
              >
                VA
              </span>
            </label>
          </div>
        </div>
        <div
          className="filter-section"
        >
          <div
            className="filter-section-title"
          >
            Type
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={true}
              className="bx--checkbox"
              id="typeall"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="typeall"
              title={null}
            >
              <span
                className=""
              >
                All
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="typeEnforce"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="typeEnforce"
              title={null}
            >
              <span
                className=""
              >
                Enforce
              </span>
            </label>
          </div>
          <div
            className="bx--form-item bx--checkbox-wrapper"
          >
            <input
              checked={false}
              className="bx--checkbox"
              id="typeInform only"
              onChange={[Function]}
              onKeyPress={[Function]}
              type="checkbox"
            />
            <label
              className="bx--checkbox-label filter-section-checkbox"
              htmlFor="typeInform only"
              title={null}
            >
              <span
                className=""
              >
                Inform only
              </span>
            </label>
          </div>
        </div>
        <div
          className="filter-section"
        >
          <div
            className="filter-section-title"
          />
        </div>
      </div>
      <div
        style={
          Object {
            "borderRadius": 3,
            "bottom": 2,
            "display": "none",
            "height": 6,
            "left": 2,
            "position": "absolute",
            "right": 2,
          }
        }
      >
        <div
          style={
            Object {
              "backgroundColor": "rgba(0,0,0,.2)",
              "borderRadius": "inherit",
              "cursor": "pointer",
              "display": "block",
              "height": "100%",
              "position": "relative",
            }
          }
        />
      </div>
      <div
        style={
          Object {
            "borderRadius": 3,
            "bottom": 2,
            "display": "none",
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 6,
          }
        }
      >
        <div
          className="filter-sections-scrollbar"
          style={
            Object {
              "backgroundColor": "rgba(255,255,255,.2)",
              "borderRadius": "inherit",
              "cursor": "pointer",
              "display": "block",
              "position": "relative",
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`ResourceToolbar component renders with no filters as expected 1`] = `
<div
  className="resource-toolbar"
  id="resource-toolbar"
>
  <div
    className="resource-toolbar-container"
  >
    <div
      className="resource-toolbar-buttons"
    >
      <div
        className="refresh-time-selection"
      >
        <div
          aria-label="Refresh"
          className="button"
          onClick={[Function]}
          onKeyPress={[Function]}
          role="button"
          tabIndex={0}
          title="Refresh"
        >
          <svg
            className="button-icon"
          >
            <use
              href="#diagramIcons_autoRefresh"
            />
          </svg>
        </div>
        <div
          aria-label="Choose refresh"
          className="bx--dropdown selection bx--list-box"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="listbox"
          tabIndex={null}
        >
          <div
            aria-controls={null}
            aria-expanded={false}
            aria-haspopup={true}
            aria-label="Choose refresh"
            aria-owns={null}
            className="bx--list-box__field"
            data-toggle={true}
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex="0"
            type="button"
          >
            <span
              className="bx--list-box__label"
              htmlFor="downshift-1-input"
            >
              Refresh every {0}s
            </span>
            <div
              className="bx--list-box__menu-icon"
            >
              <svg
                alt="Open menu"
                aria-label="Open menu"
                fillRule="evenodd"
                focusable="false"
                height="5"
                role="img"
                viewBox="0 0 10 5"
                width="10"
              >
                <title>
                  Open menu
                </title>
                <path
                  d="M0 0l5 4.998L10 0z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
